horace_core/utilities/@sqw/combine_sqw.m

```diff
@@ -1,209 +1,149 @@
-min_orig1 = cell(1,ndims1);
-max_orig1 = cell(1,ndims1);
-min_orig2 = cell(1,ndims1);
-max_orig2 = cell(1,ndims1);
-for i=1:ndims1
-    min_orig1{i}=min(w1.data.p{i});
-    max_orig1{i}=max(w1.data.p{i});
-    min_orig2{i}=min(w2.data.p{i});
-    max_orig2{i}=max(w2.data.p{i});
+right_type = arrayfun(@(x)is_sqw_type(x),inputs);
+if ~all(right_type)
+    error('HORACE:combine_sqw:invalid_argumen',...
+        'input objects must be sqw type with detector pixel information');
 end
-%NB changed index of cells in above for-loop from w1.data.pax(i) to i (15/12/15)
-
-for i=1:ndims1
-    min_full{i}=min([min_full{i} min_orig1{i} min_orig2{i}]);
-    max_full{i}=max([max_full{i} max_orig1{i} max_orig2{i}]);
+% Ignore empty datasets
+is_empty = arrayfun(@(x)(x.data.pix.num_pixels == 0),inputs);
+if all(is_empty)
+    wout= w1;
+    return;
 end
-
-%The output will eventually be an sqw object, but for now we need it to be
-%a structure array, since during intermediate steps it will not be
-%consistent with the sqw format.
-wout=get(w1);
-
-for i=1:ndims1
-    step=wout.data.p{i}(2)-wout.data.p{i}(1);
-    wout.data.p{i}=make_const_bin_boundaries([min_full{i},step,max_full{i}]);
-    % TODO: the previous value used here is different from the value,
-    % defined by make_const_bin_boundaries. All seems work. 
-    % Was those a bug or a feature?
-    %[[min_full{i}-step+eps:step:max_full{i}+step-eps]';
-end
-
-%Question - how do we deal with the fact that the integration range along
-%one or more of the non-plot axes must be different for the two objects? Do
-%we just take the average?? If we do not address this point then the
-%integration range required by w1 will be applied to the data from w2. This
-%effectively means that we are only splicing data together (e.g. sticking 2
-%2-d planes together).
-
-%We must ensure that no detector pixels are double counted when we
-%combine! Can use the function "unique" to do this:
-pixfull=[w1.data.pix.data w2.data.pix.data]';%(n1+n2)-by-9 array
-pixfull=unique(pixfull,'rows');%keeps only non-repeated rows
-
-%Now make this unique set of combined pixels the output pix array:
-wout.data.pix=PixelData(pixfull');
-
-%We need to fiddle the integration ranges so that all of the data for the
-%combined dataset is included. Can do this by looking at the minima and
-%maxima of the corresponding values in the pix array.
-nints=4-ndims1;
-intmin_l = cell(2,nints);
-intmax_l = cell(2,nints);
-intmax   = cell(1,nints);
-intmin   = cell(1,nints);
-if nints>=1
-    for i=1:nints
-        intmin_l{1,i}=min(coords_rlu1(w1.data.iax(i),:));
-        intmax_l{1,i}=max(coords_rlu1(w1.data.iax(i),:));
-        intmin_l{2,i}=min(coords_rlu2_new(w2.data.iax(i),:));
-        intmax_l{2,i}=max(coords_rlu2_new(w2.data.iax(i),:));
-    end
-    for i=1:nints
-        intmin{i}=min([intmin_l{1,i} intmin_l{2,i}]);
-        intmax{i}=max([intmax_l{1,i} intmax_l{2,i}]);
+if any(is_empty)
```

Comment:
Do we need this in an if? If none are empty, then we'll select all elements, surely.

Reply:
may be unnecessary optimization. If the array is big, it may have a difference. For realistic inputs -- probably does not make any noticeable difference